com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderUtil.java

```diff
@@ -498,7 +511,20 @@ public static Result transform(String input, ShaderDesc.ShaderType shaderType, S
                             }
 
                             if (isOpaque) {
-                                line = layout + " " + line;
+                                Matcher combinedSamplerMatcher = regexCombinedSamplerPattern.matcher(type);
+                                if(combinedSamplerMatcher.find()) {
+                                    String lines = "";
+                                    String samplerType = combinedSamplerMatcher.group("type");
+                                    lines += layout + " " + line.replaceAll("\\b" + type + "\\b", "texture" + samplerType) + System.lineSeparator();
+                                    lines += layout + " " + line.replaceAll("\\b" + type + "\\b", "sampler").replaceAll("\\b" + identifier + "\\b", identifier + "_separated") + System.lineSeparator();
+                                    line = lines;
+                                    String[] texture_replacement = {String.format("texture(\\W?)+\\((\\W?)+%s(\\W?)+,", identifier), String.format("texture(%s(%s, %s_separated),", type, identifier, identifier)};
+                                    lineReplacements.add(texture_replacement);
+                                    String[] textureLod_replacement = {String.format("textureLod(\\W?)+\\((\\W?)+%s(\\W?)+,", identifier), String.format("textureLod(%s(%s, %s_separated),", type, identifier, identifier)};
+                                    lineReplacements.add(textureLod_replacement);
```

Comment:
I put it behind the "useLatestFeatures" which seemed like it was used in the path for spirv creation which I think should work fine. This just forces separated sampler/texture (rather than a single combined), which wgsl requires and vulkan supports. If we don't do it at spirv generation then I will have to do another pass to get to spirv and then I fear that I'll need my own reflection data too. Would you prefer that?

Reply:
Ah no, that's fine I think. As long as it doesn't affect the GLSL shaders it should be fine. Have you tried the vulkan renderer with this enabled? If it works then it's fine

Reply:
I just tried with vulkan renderer and it still looks good to me.